File: inventario_frontend/src/models/sale.ts
import { Client } from "./client";
import { paymentTypes } from "@/enums/paymentTypes";
import { DetailForCreation, Detail } from "./detail";
import {
  OwnFeesForCreation,
  CashForCreation,
  CreditCardForCreation,
  DebitCardForCreation,
  ChequesPaymentForCreation,
  OwnFees,
  Cash,
  CreditCard,
  DebitCard,
  ChequesPayment
} from "./payment";

export class SaleForCreation {
  public clientId: string | undefined = undefined;
  public client: Client | undefined = undefined;
  public clientName: string | undefined = undefined;
  public date: Date | undefined = undefined;
  public paymentType: paymentTypes | undefined = undefined;
  public paymentId: string | undefined = undefined;
  public ownFees?: OwnFeesForCreation;
  public cash?: CashForCreation;
  public creditCard?: CreditCardForCreation;
  public debitCard?: DebitCardForCreation;
  public cheques?: ChequesPaymentForCreation;
  public details: DetailForCreation[] = [];
}

export class Sale extends SaleForCreation {
  public id!: string;
  public ownFees?: OwnFees;
  public cash?: Cash;
  public creditCard?: CreditCard;
  public debitCard?: DebitCard;
  public cheques?: ChequesPayment;
  public details: Detail[] = [];
}

File: Inventario/DataAccess/Models/Sale.cs
using Data.Models;
using System;
using System.Collections.Generic;
using System.Text;
using Util.Enums;

namespace Data
{
    public class Sale: Entity
    {
        public Guid? ClientId { get; set; }
        public Client Client { get; set; }
        public string ClientName { get; set; }
        public DateTime Date { get; set; }
        public ePaymentTypes PaymentType { get; set; }
        public Guid PaymentId { get; set; }
        public Payment Payment { get; set; }
        public List<Detail> Details { get; set; }
        public Sale()
        {
            Details = new List<Detail>();
        }
    }
}
